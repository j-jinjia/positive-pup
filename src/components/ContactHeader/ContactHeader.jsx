import "./ContactHeader.scss";
import mailIcon from "../../assets/svgs/mail.svg";
import Header from "../Header/Header";

const ContactHeader = ({ headingText, subheadingText, isLeftAlign }) => {
  return (
    <div className="contact-header">
      <Header
        headingText={headingText}
        subheadingText={subheadingText}
        isLeftAlign={isLeftAlign}
      />

      <p className="contact-header__text">
        Please fill in the form or alternatively, drop me an email using the
        link below.
      </p>
      <a
        className="contact-header__anchor"
        href="mailto:info@thepositivepup.co.uk"
      >
        <img src={mailIcon} alt="" />
        <p className="contact-header__button-text">info@thepositivepup.co.uk</p>
      </a>
    </div>
  );
};

export default ContactHeader;
