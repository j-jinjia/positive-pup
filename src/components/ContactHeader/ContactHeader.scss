@use "../../styles/variables" as *;

.contact-header {
  max-width: 386px;
  &__title {
    color: $colour-primary;
    font-size: $font-size-h1;
    font-family: $font-general;
    margin-bottom: 8px;
  }
  &__heading {
    font-size: $font-size-h2;
    font-family: $font-header;
    font-weight: normal;
    margin-bottom: 28px;
  }
  &__text {
    font-size: 1rem;
    line-height: 1.5;
    margin-bottom: 40px;
    color: #6f6682;
  }
  &__anchor {
    text-decoration: none;
  }
  &__button {
    display: flex;
    justify-content: space-between;
    align-items: center;
    padding: 20px;
    width: 270px;
    background: transparent;
    color: $colour-primary;
    border: none;
    border-radius: 14px;
    box-shadow: 0px 20px 40px rgba(55, 175, 162, 0.08);
    &-text {
      text-decoration: none;
    }
    &:hover {
      cursor: pointer;
    }
  }
}

@media screen and (min-width: $desktop-breakpoint) {
  .contact-header {
    &__heading {
      font-size: $font-size-h2-desktop;
    }
  }
}
