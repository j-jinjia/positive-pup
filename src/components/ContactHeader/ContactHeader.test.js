import { customRender } from "../../utils/testUtils";
import ContactHeader from "./ContactHeader";

it("should match ContactHeader component snapshot", () => {
  const { container } = customRender(<ContactHeader />);

  expect(container).toMatchSnapshot();
});
