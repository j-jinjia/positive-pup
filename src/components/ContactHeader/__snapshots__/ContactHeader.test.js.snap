// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match ContactHeader component snapshot 1`] = `
<div>
  <div
    class="contact-header"
  >
    <div
      class="header"
    >
      <h1
        class="header__heading"
      >
         
         
      </h1>
      <h2
        class="header__subheading"
      />
      <h2
        class="header__subheading"
      />
    </div>
    <p
      class="contact-header__text"
    >
      Please fill in the form or alternatively, drop me an email using the link below.
    </p>
    <a
      class="contact-header__anchor"
      href="mailto:info@thepositivepup.co.uk"
    >
      <img
        alt=""
        src="mail.svg"
      />
      <p
        class="contact-header__button-text"
      >
        info@thepositivepup.co.uk
      </p>
    </a>
  </div>
</div>
`;
